### ISSUE TITLE
Remote node CRD

### ISSUE THREAD

**Issue by zlobober** [ytsaurus/yt-k8s-operator#75]
We'd like to have an opportunity to describe a group of nodes (dat/exe/tab) as a separate CRD, providing the endpoints of a remote cluster. Such group of nodes (called remote nodes) will work the same as if they were described within the ytsaurus CRD, but they can belong to a different k8s cluster, which may be useful.

I'd suggest having a CRD RemoteNodeGroup, which looks exactly the same as dataNodes/execNodes/tabletNodes (and, ideally, reuses as much of existing code as possible). 

The main difference is that there should be a field "remoteClusterSpec" which must be a reference to a resource (CRD) of type RemoteYtsaurusSpec.

RemoteYtsaurusSpec must describe necessary information for connecting a node to a remote cluster. remoteClusterSpec must be a structure, currently containing one field:
```
MasterAddresses: []string
```

But in the future it will probably be extended by various other fields.
When https://github.com/ytsaurus/ytsaurus/issues/248 completes:
```
RemoteClusterUrl: string // allows taking cluster connection by HTTP cal to RemoteClusterUrl + "/cluster_connection".
```

When authentication in native protocol is employed:
```
Credentials: SomeKindOfCredentials
```

**Comment by psushin** [comment]
Should RemoteYsaurusSpec be a separate CRD, or just described in a config map?
This CRD doesn't have any behavior, as far as I can see, so no particular controller, reconciliation, etc.

**Comment by zlobober** [comment]
We discussed this matter with k8s-experienced colleagues, and it seems like having CRD for such case is a common practice. Configmap is unschematized object, and CRD is schematized, which is much less error-prone. In general, CRD does not necessarily require a designated controller, it may be a part of a more complex CRD like in our case.

**Comment by l0kix2** [comment]
This change will be separated in a chain of PRs to ease codereviewing.
0. (merged) #83 in which we increase test coverage of config generator so we don't break things in following refactoring;
1. #84 in which config generator is being refactored to be able to work with remote nodes in the future.
2. (WIP) server/statefulset refactoring to be able to work without full ytsaurus spec with only master connection
3. (WIP) new CRDs for remote nodes & remote ytsaurus + reconcillers

**Comment by l0kix2** [comment]
> This CRD doesn't have any behavior, as far as I can see, so no particular controller, reconciliation, etc.

> In general, CRD does not necessarily require a designated controller, it may be a part of a more complex CRD like in our case.

To be clear, `RemoteYsaurus` CRD doesn't require reconciliation controller, but if master hosts in `RemoteYsaurus` are changed — `RemoteNodes` controller should update nodes configs. Therefore `RemoteNodes` controller need to watch connected `RemoteYtsaurus` changes. 
Implementation-wise it would be simpler to inline(or yaml-anchor-reference from the same file) masterHosts+cellTag in `RemoteNodes` spec, though I've found [an example ](https://book.kubebuilder.io/reference/watching-resources/externally-managed) of watching external resource from controller — it requires a bit of boilerplate code, but it seems to work, so we can proceed with separate CRD option.
